20200120-drivetrain merge with master

LeviWalker opened the pull request:
Merge 20200120-drivetrain into 20200130-drivetrain

LeviWalker committed:
Merge branch '20200120-drivetrain' of C:\Users\kids\IdeaProjects\InfiniteRecharge with conflicts.

LeviWalker committed:
Drivetrain updates

LeviWalker committed:
Update Turn PID values with the SmartDashboard application

LeviWalker committed:
Initial Shooter subsystem commit

LeviWalker committed:
Drivetrain subsystem rewrite and beginnings of distance PID controller

LeviWalker committed:
Added pneumatics to the drivetrain and CAN IDs

LeviWalker committed:
Added some JavaDoc documentation comments for the drivetrain and the commands that implement the drivetrain's methods

bhooten committed:
Merge branch 'master' into 20200120-drivetrain

bhooten committed:
Update and resolve conflicts

bhooten committed:
Resolve broken references

bhooten commented on src/main/java/xyz/nasaknights/infiniterecharge/commands/drivetrain/VisionDriveAssistCommand.java:
```diff
@@ -0,0 +1,36 @@
+public class VisionDriveAssistCommand extends CommandBase
+{
+    public VisionDriveAssistCommand()
+    {
```
```suggestion
    {
        addRequirements(RobotContainer.getDrivetrain());
```

bhooten commented on src/main/java/xyz/nasaknights/infiniterecharge/commands/drivetrain/VisionDriveAssistCommand.java:
```diff
@@ -0,0 +1,36 @@
+package xyz.nasaknights.infiniterecharge.commands.drivetrain;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
```
```suggestion
import xyz.nasaknights.infiniterecharge.RobotContainer;
```

bhooten committed:
Update src/main/java/xyz/nasaknights/infiniterecharge/commands/drivetrain/VisionDriveAssistCommand.java

bhooten committed:
Update src/main/java/xyz/nasaknights/infiniterecharge/commands/drivetrain/VisionDriveAssistCommand.java

bhooten reviewed (changes requested):
Changes need to be tested before committing

LeviWalker committed:
Merge branch '20200120-drivetrain' of C:\Users\kids\IdeaProjects\InfiniteRecharge with conflicts.

LeviWalker committed:
Testing updates

LeviWalker commented:
This code has been tested on the robot and is ready to merge.

LeviWalker committed:
Flip GTA Turn value

The turn value for the GTA case was multiplied by negative one after testing to make the robot turn the direction of the joystick axis.

LeviWalker committed:
Flip GTA Turn value

The turn value for the GTA case was multiplied by negative one after testing to make the robot turn the direction of the joystick axis.

LeviWalker committed:
Edit GTA drive case movement

Flipped the accelerate and decelerate triggers and moved the turn axis to the x-axis of the left joystick.

LeviWalker committed:
Add reversed GTA case

Added the `GTA_REVERSED` case to the `ControlType` enumerator.

LeviWalker committed:
Add GTA_REVERSED case movement

Mapped the `GTA_REVERSED` case movement in `DriveCommand`.

bhooten commented:
This branch is stale and was superseded in PR #18. Closed.
